packages/server/src/chat/service.ts

```diff
@@ -0,0 +1,120 @@
+import { Message, uuid } from '@botpress/messaging-base'
+import _ from 'lodash'
+import yn from 'yn'
+import { Service } from '../base/service'
+import { ChannelService } from '../channels/service'
+import { ClientService } from '../clients/service'
+import { ConduitService } from '../conduits/service'
+import { ConfigService } from '../config/service'
+import { ConversationService } from '../conversations/service'
+import { InstanceService } from '../instances/service'
+import { LoggerService } from '../logger/service'
+import { Logger } from '../logger/types'
+import { MappingService } from '../mapping/service'
+import { MessageService } from '../messages/service'
+import { PostService } from '../post/service'
+import { SocketService } from '../socket/service'
+import { WebhookBroadcaster } from '../webhooks/broadcaster'
+import { WebhookService } from '../webhooks/service'
+import { WebhookContent } from '../webhooks/types'
+
+export class ChatService extends Service {
+  private webhookBroadcaster: WebhookBroadcaster
+  private loggingEnabled!: boolean
+  private logger!: Logger
+
+  constructor(
+    private loggers: LoggerService,
+    private configs: ConfigService,
+    private posts: PostService,
+    private channels: ChannelService,
+    private clients: ClientService,
+    private webhooks: WebhookService,
+    private conduits: ConduitService,
+    private conversations: ConversationService,
+    private messages: MessageService,
+    private mappings: MappingService,
+    private instances: InstanceService,
+    private sockets: SocketService
+  ) {
+    super()
+    this.webhookBroadcaster = new WebhookBroadcaster(this.posts, this.webhooks)
+    this.logger = this.loggers.root.sub('chat')
+  }
+
+  async setup() {
+    if (process.env.LOGGING_ENABLED?.length) {
+      this.loggingEnabled = !!yn(process.env.LOGGING_ENABLED)
+    } else if (this.configs.current.logging?.enabled !== null && this.configs.current.logging?.enabled !== undefined) {
+      this.loggingEnabled = !!yn(this.configs.current.logging.enabled)
+    } else {
+      this.loggingEnabled = process.env.NODE_ENV !== 'production'
+    }
+  }
+
+  async send(conversationId: uuid, authorId: uuid | undefined, payload: any, from: any): Promise<Message> {
+    const conversation = await this.conversations.get(conversationId)
+    const client = await this.clients.getById(conversation!.clientId)
```

Comment:
In that case it would be an incorrect usage of the function and an exception would be thrown on the second line. The check to see if it's a valid conversation is done at the route level, no need to put checks everywhere.